tests/internals/pause_scaledobject_explicitly/pause_scaledobject_explicitly_test.go

```diff
@@ -0,0 +1,202 @@
+
+	// test scaling
+	testPauseWhenScaleOut(t, kc)
+	testScaleOut(t, kc)
+	testPauseWhenScaleIn(t, kc)
+	testScaleIn(t, kc)
+
+	// cleanup
+	DeleteKubernetesResources(t, testNamespace, data, templates)
+}
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+			TestNamespace:           testNamespace,
+			DeploymentName:          deploymentName,
+			ScaledObjectName:        scaledObjectName,
+			MonitoredDeploymentName: monitoredDeploymentName,
+		}, []Template{
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+			{Name: "monitoredDeploymentTemplate", Config: monitoredDeploymentTemplate},
+			{Name: "scaledObjectAnnotatedTemplate", Config: scaledObjectTemplate},
+		}
+}
+
+func upsertScaledObjectAnnotation(t assert.TestingT) {
+	_, err := ExecuteCommand(fmt.Sprintf("kubectl annotate scaledobject/%s -n %s autoscaling.keda.sh/paused='true' --overwrite", scaledObjectName, testNamespace))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+}
+
+func removeScaledObjectAnnotation(t assert.TestingT) {
+	_, err := ExecuteCommand(fmt.Sprintf("kubectl annotate scaledobject/%s -n %s autoscaling.keda.sh/paused- --overwrite", scaledObjectName, testNamespace))
+	assert.NoErrorf(t, err, "cannot execute command - %s", err)
+}
+
+func testPauseWhenScaleOut(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing pausing at 0 ---")
+
+	upsertScaledObjectAnnotation(t)
+	KubernetesScaleDeployment(t, kc, monitoredDeploymentName, 2, testNamespace)
+	assert.Truef(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, 2, 60, testScaleOutWaitMin),
+		"monitoredDeploymentName replica count should be 2 after %d minute(s)", testScaleOutWaitMin)
+
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 0, 10)
+}
+
+func testScaleOut(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing scale out ---")
+
+	removeScaledObjectAnnotation(t)
+
+	KubernetesScaleDeployment(t, kc, monitoredDeploymentName, 5, testNamespace)
+	assert.Truef(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, 5, 60, testScaleOutWaitMin),
+		"monitoredDeploymentName replica count should be 5 after %d minute(s)", testScaleOutWaitMin)
+
+	assert.Truef(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 5, 60, testScaleOutWaitMin),
+		"replica count should be 5 after %d minute(s)", testScaleOutWaitMin)
+}
+
+func testPauseWhenScaleIn(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing pausing at N ---")
+
+	KubernetesScaleDeployment(t, kc, monitoredDeploymentName, 5, testNamespace)
+
+	assert.Truef(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 5, 10, testPauseAtNWaitMin),
+		"replica count should be 5 after %d minute(s)", testPauseAtNWaitMin)
+
+	upsertScaledObjectAnnotation(t)
+
+	KubernetesScaleDeployment(t, kc, monitoredDeploymentName, 0, testNamespace)
```

Comment:
Before we do this, can we check that it's still at 5 replicas?

Reply:
Or you prefer doing after when it scales to 0 so that it should have scaled in if it was active, but is still at 5? If so, fine for me.

Reply:
Added checking to make sure it's still at 5 replicas